File: ml/python/tts/tts_openai.py
from openai import OpenAI

class TtsOpenAI:
    def __init__(self, api_key):
        self.client = OpenAI(api_key=api_key)

    def speak(self, text):
        response = self.client.audio.speech.create(
            model="tts-1",
            voice="alloy",
            input=text
        )
        return response.content

File: backend/ml/js/tts/ttsOpenAI.js
import OpenAI from "openai";

class TtsOpenAI {
    constructor(apiKey) {
        this.openai = new OpenAI({ apiKey });
    }

    async speak(text) {
        const mp3 = await this.openai.audio.speech.create({
            model: "tts-1",
            voice: "alloy",
            input: text,
        });
        return Buffer.from(await mp3.arrayBuffer());
    }
}

export default TtsOpenAI;
